docs/xroad_security_server_toolkit_user_guide.md

```diff
@@ -106,17 +111,111 @@ $ pip install -r requirements-dev.txt
+
+__Proper care must be taken to ensure that configuration files with these credentials are not visible
+to strangers' eyes, as the secrets they can contain (API keys, administrative credentials) are
+stored in plain text. Possible information leaks are minimized when using API keys, without any SSH
+connections or security server administrative credentials configured.__
+
+Security server REST API is ordinarily exposed at security server port 4000 and is separated into
+two parts:
+  1. invocable over network -- API calls for performing most of the functionality available from
+     web administration console, accessible with API key.
+  1. invocable only locally (in default configuration), i.e. when accessed via 'localhost' or
+     equivalent and passed security server administrative credentials via HTTP basic access
+     authentication -- API calls that allow API key management operations.
+
+##### 3.1.1.1 Using API keys
+
+The API key used by toolkit against REST API needs following roles for full toolkit
+functionality:
+ 1. XROAD_SYSTEM_ADMINISTRATOR
+ 1. XROAD_SERVICE_ADMINISTRATOR
+ 1. XROAD_SECURITY_OFFICER
+ 1. XROAD_REGISTRATION_OFFICER
+
+On freshly installed and completely unconfigured security server, API key can be obtained from
+the server with local API invocation, e.g.:
+```sh
+$ curl -k --silent \
+    -X POST \
+    --header 'Content-Type: application/json' \
+    --data '["XROAD_SYSTEM_ADMINISTRATOR", "XROAD_SERVICE_ADMINISTRATOR"]' \
+    -u user:pass https://localhost:4000/api/v1/api-keys \
+ | jq
+{
+   "id":122,
+   "key":"55f0a0ce-46d6-4217-a15a-43e026c4a9c5",
+   "roles":["XROAD_SYSTEM_ADMINISTRATOR","XROAD_SERVICE_ADMINISTRATOR"]
+}
+```
+
+In the above sample command, role list was shortened for readability, ``user:pass`` to use are
+X-Road security server administrative access credentials. In the output sample,
+``key`` is the UUID to be used as REST API key. This UUID is only retrievable once for
+API key, as it is not stored in plaintext at the security server. If dealing with security
+server that already has some basic configuration done (anchors, ownership information
+and software token configuration finished), this API key can also be created from the web
+administration console (from "Keys and certificates" -> "API keys" submenu).
+
+##### 3.1.1.2 Using SSH
+
+For ease of automation and development experiments, toolkit is also able to perform operations
+when not supplied with API key, but given SSH access credentials to configurable server and security
+server administration credentials. In this case, it will create transient API keys for performing
+the configuration operations, in the same way as described above, and these API keys are normally
+revoked when the toolkit command finishes. However, in case of e.g. electricity or network
+connection loss these keys could remain on the server indefinitely.
```

Comment:
The server where the keys can remain - is it the toolkit host or the security server or perhaps both?

Reply:
Toolkit host has them only in memory, not in permanent storage, so that would be security server only, changed to:

> could remain on the **security server** indefinitely.